woocommerce/woocommerce
# Fix PHP notice when outside of orders screens

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





After merging #38617 I _noticed_ a PHP notice started appearing on all screens:

<img width="1028" alt="Screenshot 2023-06-07 at 16 15 49" src="https://github.com/woocommerce/woocommerce/assets/184724/6d580075-af0f-4b47-939c-f271686c2f33">

While #38617 made this evident, this is IMHO a failure in our orders page controller, since its logic is executed on every page even when not viewing an orders-related screen.

This PR addresses this by returning early if we're not on an admin page with `wc-orders` in its slug.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Check out `trunk`.
1. Visit any page outside of the Orders screen.
1. Confirm that you see the PHP notice mentioned above:
   ```Function Automattic\WooCommerce\Internal\Admin\Orders\PageController::is_order_screen was called incorrectly. is not a valid order type
   ```
1. Visit WC > Orders and confirm that the notice no longer appears.
1. Repeat the above steps on this branch, and confirm that the notice does not appear at any point.
